connector_spec/connector_spec_ui.ts

```diff
@@ -0,0 +1,242 @@
+      /**
+       * Mark field as sensitive (password/token/secret)
+       * WHY: Can't infer from z.string() whether value should be masked
+       * EFFECT: Renders as password input, masked in logs/errors
+       * @example z.string().meta({ sensitive: true })
+       */
+      sensitive?: boolean;
+      
+      /**
+       * Override default widget for this field type
+       * WHY: Same Zod type can need different UI widgets
+       * - z.string() could be: text input, textarea, JSON editor, code editor
+       * - z.record() could be: key-value editor, JSON editor
+       * @example z.string().meta({ widget: "textarea" }) // Multi-line
+       * @example z.string().meta({ widget: "json" }) // JSON editor with syntax highlighting
+       */
+      widget?: 
+        | "text"          // Single-line text input (default for z.string())
+        | "textarea"      // Multi-line text input
+        | "password"      // Masked password input (auto-applied if sensitive: true)
+        | "json"          // JSON editor with syntax highlighting and validation
+        | "code"          // Code editor (specify language in widgetOptions)
+        | "keyValue"      // Key-value pair editor (for z.record())
+        | "number"        // Number input with increment/decrement (default for z.number())
+        | "select"        // Dropdown (auto-applied for z.enum() and when optionsFrom is set)
+        | "multiSelect"   // Multi-select dropdown (for z.array(z.enum()))
+        | "toggle"        // Toggle switch (default for z.boolean())
+        | "date"          // Date picker
+        | "dateTime";     // Date and time picker
+      
+      /**
+       * Widget-specific configuration options
+       * WHY: Different widgets need different configuration
+       * @example { language: "javascript" } for code widget
+       * @example { rows: 10 } for textarea widget
+       */
+      widgetOptions?: {
+        language?: "javascript" | "typescript" | "json" | "yaml" | "python";
+        rows?: number;
+        cols?: number;
+        [key: string]: unknown;
+      };
+      
+      /**
+       * Placeholder text shown in empty input
+       * WHY: Placeholder provides example/hint, different from label
+       * Label = "API Token", Placeholder = "xoxb-1234-5678-..."
+       * @example z.string().meta({ placeholder: "https://api.example.com" })
+       */
+      placeholder?: string;
+      
+      /**
+       * Section/group this field belongs to
+       * WHY: Long forms need grouping for usability
+       * Fields with same section value are grouped together in UI
+       * @example z.string().meta({ section: "Authentication" })
+       */
+      section?: string;
+      
+      /**
+       * Explicit display order (lower = shown first)
+       * WHY: Object property order in JS/TS isn't always guaranteed
+       * Without this, fields might render in any order
+       * @example z.string().meta({ order: 1 })
+       */
+      order?: number;
```

Comment:
If a section is provided, should the ordering apply only within that section? And how should the sections themselves be ordered?